core/src/main/scala/org/http4s/otel4s/middleware/ClientMiddleware.scala

```diff
@@ -123,45 +123,39 @@ object ClientMiddleware {
               .build
               .resource
             span = res.span
-            traceHeaders <- Resource.eval(Tracer[F].propagate(Headers.empty))
+            trace = res.trace
+            traceHeaders <- Resource.eval(trace(Tracer[F].propagate(Headers.empty)))
```

Comment:
it would probably be cleaner to use `mapK`, as follows:

```suggestion
            traceHeaders <- Resource.eval(Tracer[F].propagate(Headers.empty)).mapK(trace)
```